plotter/views.py

```diff
@@ -818,14 +846,190 @@ def export(request):
+            fun = getattr(pro, datatype)
+            # need to
+            dicts = fun(authenticated_userid(request))
+            pinfo["table"] = datatype
+            pinfo["visibility"] = EXPORT_VISIBILITY[datatype]
+            for d in dicts:
+                d["user_id"] = authenticated_userid(request)
+                d["profile_id"] = x
+            if datatype != "segments":
+                tosplitbed = respond_bed_csv(datatype, "bed", pinfo, dicts, False).text
+            else:
+                tosplitbed = respond_bed_csv(datatype, "bedGraph", pinfo, dicts, False).text
+            if tosplitbed == '':
+                continue
+            tosplitbedlist = tosplitbed.split('\n')
+            tosortbedlist = []
+            for line in tosplitbedlist:
+                if line != '':
+                    tosortbedlist.append(line.split(' '))
+            tounsplitbedlist = sorted(tosortbedlist, key=lambda lambdaline: (unicode(lambdaline[0]), int(lambdaline[1])))
+            unsplitbedlist = []
+            for line in tounsplitbedlist:
+                unsplitbedlist.append('\t'.join(line))
+            tosavebed = '\n'.join(unsplitbedlist)
+            if datatype != "segments":
+                bedfile = open(x+"_"+datatype+'.bed', 'w')
+            else:
+                bedfile = open(x+"_"+datatype+'.bedGraph', 'w')
+            bedfile.write(tosavebed)
+            bedfile.close()
+            usedlist.append(datatype)
+            if datatype != "segments":
+                subprocess.call(['bedToBigBed', x+"_"+datatype+'.bed', 'chrom.sizes', x+"_"+datatype+'.bigbed'])
+            else:
+                subprocess.call(['bedGraphToBigWig', x + "_segments.bedGraph", 'chrom.sizes', x + "_segments.bigWig"])
+            if x != "ES0004" and datatype != "segments":
+                os.remove(x+"_"+datatype+'.bed')
+            elif x != "ES0004":
+                os.remove(x+"_segments.bedGraph")
+        bedgraphsecretfilelocation = db.secret_file(x+'.bedGraph.gz')
+        copy2(bedgraphsecretfilelocation, os.getcwd()+'/'+x+'.bedGraph.gz')
+        gzipfile = gzip.open(os.getcwd()+'/'+x+'.bedGraph.gz')
+        gzipfile.next()
+        bedgraphdata = gzipfile.read()
+        gzipfile.close()
+        os.remove(os.getcwd()+'/'+x+'.bedGraph.gz')
+        bedgraph = open(x+'.bedGraph', 'w')
+        bedgraph.write(bedgraphdata)
+        bedgraph.close()
+        subprocess.call(['bedGraphToBigWig', x+'.bedGraph', 'chrom.sizes', x+'.bigWig'])
+        os.remove(x+'.bedGraph')
+    trackdbtxt = open('trackDb.txt', 'w')
+    for x in request.POST.getall('profile'):
+        trackdbtxt.write('track %s' % x + 'multiWig\ntype bigWig\ncontainer multiWig\naggregate transparentOverlay\n' +
+                         'shortLabel ' + request.POST['short_label'] + 'multiWig\nlongLabel ' +
+                         request.POST['long_label'] + 'multiWig\nautoScale on\nnegativeValues on\nalwaysZero on\n\n')
```

Comment:
sorry for not being clear earlier -- I meant to say that it would be good to replace all of the + with % (not just the first)

In fact I think it would be even easier to understand/read if you use triple quotes and % here, e.g.

```python
    trackdbtxt.write("""track %smultiWig
type bigWig
shortLabel %smultiWig
"""%(x, request.POST['short_label'])
```
here i just show what it should look like with two variable substitutions in the text, but please use %s/etc (not +) for all variable substitutions

the only difference between single and triple quotes is that triple quotes allow line breaks in literal character strings... which is very useful from a readability perspective here